### PR TITLE
make sure we're not using relative paths when loading files

### PR DESCRIPTION
**Pull request by ZacLN**

### CODE REVIEW

File: src/StaticLint.jl [hunk 1]
```diff
@@ -144,11 +144,11 @@ function followinclude(x, state::State)
         path = get_path(x, state)
         if isempty(path)
         elseif hasfile(state.server, path)
-        elseif canloadfile(state.server, path)
+        elseif isabspath(path) && canloadfile(state.server, path)
```
Comment (davidanthoff) [inline src/StaticLint.jl:147]: Wouldn't it make more sense to call `isabspath` right after the call to `get_path`, and then make sure we for example don't even call `hasfile` on a relative path? So essentially we only try the options where we `joinpath` with something if `get_path` returns a relative path.

File: src/StaticLint.jl [hunk 2]
```diff
@@ -143,11 +143,11 @@ function followinclude(x, state::State)
...
         path = get_path(x, state)
         if isempty(path)
-        elseif hasfile(state.server, path)
-        elseif isabspath(path) && canloadfile(state.server, path)
-            loadfile(state.server, path)
-        elseif hasfile(state.server, joinpath(dirname(getpath(state.file)), path))
-            path = joinpath(dirname(getpath(state.file)), path)
+        elseif isabspath(path)
+            if hasfile(state.server, path)
+            elseif isabspath(path) && canloadfile(state.server, path)
```
Comment (davidanthoff) [inline src/StaticLint.jl:148]:
```suggestion
            elseif canloadfile(state.server, path)
```
We already check for that in the condition above.

**Comment by ZacLN** [comment]
I managed to pull out the branch that checked for already loaded files using the relative path - I've added it back in

**Comment by ZacLN** [comment]
That should be handled on the LS side of things then I think, I'm happy to make a PR over there.

**Pull request merged**